blog/_posts/2021-08-09-avida-ed-evolution.md

```diff
@@ -0,0 +1,528 @@
+    if (av.debug.dnd) console.log('parents', av.parents);
+  };
+```
+
+* containerMap to replace dojo's 'map'
+```Javascript
+/*
+  Helpers for Inserting and Removing Items from Dragula Containers
+  */
+
+  av.dnd.empty = function(target) {
+    var container = target.id !== undefined ? "#" + target.id : "." + target.className;
+    $(container).empty();
+    if (Object.keys(containerMap).indexOf(container) != -1) {
+      containerMap[container] = {}
+    }
+  }
+
+  av.dnd.remove = function(target, el) {
+    var container = target.id !== undefined ? "#" + target.id : "." + target.className;
+    $(container).remove(el);
+    document.querySelector(container).removeChild(document.getElementById(el.id));
+    delete containerMap[container][el.id];
+  }
+
+  av.dnd.insert = function(target, el, type) {
+    var domid = el.id;
+    var container = target.id !== undefined ? "#" + target.id : "." + target.className;
+    if (Object.keys(containerMap).indexOf(container) === -1) {
+      containerMap[container] = {}
+    }
+    console.log(el);
+    console.log($(el));
+    containerMap[container][domid] = {'name': el.textContent.trim() , 'type': type};
+  }
+
+  av.dnd.insertToDOM = function(target, el) {
+    var container = target.id !== undefined ? "#" + target.id : "." + target.className;
+    $(container).append(el);
+  }
+
+  av.dnd.getAllItems = function (source) {
+    'use strict';
+    var container = source.id !== undefined ? "#" + source.id : "." + source.className;
+    try {
+      var items = Object.keys(containerMap[container]);
+    } catch (error) {
+      var items = [];
+    }
+    return items;
+  };
+```
+* I realize these snippets are kind of huge. I will get the essential snippets for the actual blog. :)
+
+# Moving forward
+
+My goal was to make Avida-ED 4 more user-friendly and accesible for more people. Implementing the dragbar and the new drag and drop definitely came with its challenges, but in the end I think I got so much out of the experience, beyond just what I was doing. Namely, the fact that I was working on a meaningful, and historical! software that is Avida-ED 4 and seeing my often futile attempts come to fruition on the webpage was simply pure joy. 
+
+Acknowledgements
+--
+Of course, I cannot forget to express my utmost gratitude to Diane Blackwood, without whom I would not have been able to achieve as much as I did. Talking with Diane on Zoom (and Wesley!) about not just work but also different aspects of our lives was delightful! And to all of the WAVE mentors, collegues, and Matthew!: Thank you so much for supporting me and encouraging me all this way; I really enjoyed hearing about everyone's contributions and progress throughout the workshop! It was inspiring, especially when coding wasn't going smooth, haha. So thank you. I hope our paths cross again in the future!
+
```

Comment:
Overall this blog post reads incredibly well. You have just the right level of informality-- it's fun to read but still comes off as professional. Creating something like a drag and drop seems like it should be easy but I can only imagine the logistical nightmare that ensued with some of those changes. Congrats! I honestly don't have much to suggest in ways of improvements but if you really wanted to go above and beyond you could talk a little bit more about the steps you took to create the system and some of the specific challenges and solutions you encountered. Overall, great job!